packages/aws-cdk-lib/aws-stepfunctions-tasks/lib/emr/private/cluster-utils.ts

```diff
@@ -122,12 +122,39 @@ export function InstanceTypeConfigPropertyToJson(property: EmrCreateCluster.Inst
  */
 export function InstanceFleetProvisioningSpecificationsPropertyToJson(property: EmrCreateCluster.InstanceFleetProvisioningSpecificationsProperty) {
   return {
-    SpotSpecification: {
-      AllocationStrategy: cdk.stringToCloudFormation(property.spotSpecification.allocationStrategy),
-      BlockDurationMinutes: cdk.numberToCloudFormation(property.spotSpecification.blockDurationMinutes),
-      TimeoutAction: cdk.stringToCloudFormation(property.spotSpecification.timeoutAction?.valueOf()),
-      TimeoutDurationMinutes: cdk.numberToCloudFormation(property.spotSpecification.timeoutDurationMinutes),
-    },
+    OnDemandSpecification: OnDemandProvisioningSpecificationPropertyToJson(property.onDemandSpecification),
+    SpotSpecification: SpotProvisioningSpecificationPropertyToJson(property.spotSpecification),
+  };
+}
+
+/**
+ * Render the OnDemandProvisioningSpecificationProperty to JSON
+ *
+ * @param property
+ */
+function OnDemandProvisioningSpecificationPropertyToJson(property: EmrCreateCluster.OnDemandProvisioningSpecificationProperty | undefined) {
+  if (!property) {
+    return undefined;
+  }
+  return {
+    AllocationStrategy: cdk.stringToCloudFormation(property.allocationStrategy),
+  };
+}
+
+/**
+ * Render the SpotProvisioningSpecificationProperty to JSON
+ *
+ * @param property
```

Comment:
```suggestion
```

Reply:
If you want to leave an `@param` docstring, you have to describe the property like

```
@param property the spot provisioning specification
```

(But we don't need to do that here, I dont think)